results/pkg/watcher/reconciler/controller_test.go

```diff
@@ -81,35 +93,76 @@ func newResultsClient(t *testing.T) pb.ResultsClient {
 	return pb.NewResultsClient(conn)
 }
 
-// TestReconcile tests if TaskRun in the client can be properly updated when sent to reconciler
+type UnknownResource struct {
+	metav1.ObjectMeta
+}
+
+// TestReconcile tests if PipelineRun in the client can be properly updated when sent to reconciler
 func TestReconcile(t *testing.T) {
 	client := newResultsClient(t)
 	taskRun := &v1beta1.TaskRun{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        "Tekton",
+			Name:        "Tekton-TaskRun",
 			Namespace:   "default",
 			Annotations: map[string]string{"demo": "demo"},
 			UID:         "12345",
 		},
 	}
+	pipelineRun := &v1beta1.PipelineRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        "Tekton-PipelineRun",
+			Namespace:   "default",
+			Annotations: map[string]string{"demo": "pipelinerun_demo"},
+			UID:         "54321",
+		},
+	}
+
+	unknownResource := &UnknownResource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "UnknownResource",
+			Namespace: "default",
+		},
+	}
+
+	asset, ctx := getFakeClients(t, []*v1beta1.TaskRun{taskRun}, []*v1beta1.PipelineRun{pipelineRun}, client)
+
+	t.Run("test-reconcile-unknown", func(t *testing.T) {
+		c := asset.Controller
+		err := c.Reconciler.Reconcile(ctx, strings.Join([]string{unknownResource.Namespace, unknownResource.Name}, "/"))
+		if !errors.IsNotFound(err) {
+			t.Errorf("Unexpected error: %v", err)
+		}
+	})
 
-	asset, ctx := getFakeClients(t, []*v1beta1.TaskRun{taskRun}, client)
 	// Create a new TaskRun in the fake client, completed TaskRun should be updated with a results id field in annotations.
-	t.Run("test-create", func(t *testing.T) {
+	t.Run("test-create-taskrun", func(t *testing.T) {
 		tr, err := reconcileTaskRun(ctx, asset, taskRun)
 		if err != nil {
 			t.Fatalf("Failed to get completed TaskRun %s: %v", taskRun.Name, err)
 		}
-		if _, ok := tr.Annotations[idName]; !ok {
+		if _, ok := tr.Annotations[IDName]; !ok {
 			t.Fatalf("Expected completed TaskRun %s should be updated with a results_id field in annotations", taskRun.Name)
 		}
-		if _, err := client.GetResult(ctx, &pb.GetResultRequest{Name: tr.Annotations[idName]}); err != nil {
+		if _, err := client.GetResult(ctx, &pb.GetResultRequest{Name: tr.Annotations[IDName]}); err != nil {
 			t.Fatalf("Expected completed TaskRun %s not created in api server", taskRun.Name)
 		}
 	})
 
+	t.Run("test-create-pipelinerun", func(t *testing.T) {
```

Comment:
many of these tests look very similar. Could we use a table driven test here? https://github.com/golang/go/wiki/TableDrivenTests

Reply:
No problem!